### PR TITLE
Rename Node Playground to User Scripts

### PR DESCRIPTION
**Pull request by foxymiles**
**User-Facing Changes**
Node Playground is now called `User Scripts`.

**Description**
This just changes the name displayed to the user. We could also do a bigger PR and change the names in the codebase.

### CODE REVIEW

**Comment by amacneil** [comment]
cc @esthersweon to coordinate docs

File: packages/studio-base/src/panels/NodePlayground/index.help.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Write code to manipulate, aggregate, and filter existing messages on topics and output them to new topics for other Studio panels to visualize.
 
-_Node Playground_ scripts are written in [TypeScript](https://www.typescriptlang.org/).
+_User Scripts_ scripts are written in [TypeScript](https://www.typescriptlang.org/).
```
Comment (amacneil) [inline packages/studio-base/src/panels/NodePlayground/index.help.md:3]:
```suggestion
_User Scripts_ are written in [TypeScript](https://www.typescriptlang.org/).
```

File: packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/typescript/templates/marker.ts.template [hunk 2]
```diff
@@ -1,6 +1,6 @@
-// This example shows how to publish a Marker message from a Node Playground node.
+// This example shows how to publish a Marker message from a User Scripts node.
 //
-// Publishing Marker messages with a Node Playground script is a good way to visualize non-visual
+// Publishing Marker messages with a User Scripts script is a good way to visualize non-visual
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/typescript/templates/marker.ts.template:3]:
```suggestion
// Publishing Marker messages with a User Script is a good way to visualize non-visual
```

File: packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/typescript/templates/multipleInputs.ts.template [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // This example shows how to subscribe to multiple input topics.
 //
 // NOTE:
-// Node Playground scripts can subscribe to multiple input topics, but can only publish on a single topic.
+// User Scripts scripts can subscribe to multiple input topics, but can only publish on a single topic.
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/typescript/templates/multipleInputs.ts.template:4]:
```suggestion
// User Scripts can subscribe to multiple input topics, but can only publish on a single topic.
```

**Comment by amacneil** [comment]
Also I suggest replacing instances of `User Scripts node` with just `User Script`

**Comment by defunctzombie** [comment]
image

Should this read `Create a new script` ?

**Comment by amacneil** [comment]
I think we should stop using the term node - they are scripts. Node is somewhat confusing anyway, because people think it is going to publish back to ROS.

File: packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/__snapshots__/generateRosLib.test.ts.snap [hunk 4]
```diff
@@ -40,7 +40,7 @@ exports[`typegen generateRosLib basic snapshot 1`] = `
      * Input<\\"/your_input_topic_2\\">'.
      *
      * These types are dynamically generated from the bag(s) currently in your
-     * Foxglove Studio session, so if a datatype changes, your Node Playground
+     * Foxglove Studio session, so if a datatype changes, your User Scripts
      * node may not compile on the newly formatted bag.
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/__snapshots__/generateRosLib.test.ts.snap:44]:
```suggestion
     * Foxglove Studio session, so if a datatype changes, your User Script
     * may not compile on the newly formatted bag.
```

File: packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/__snapshots__/generateRosLib.test.ts.snap [hunk 5]
```diff
@@ -332,7 +332,7 @@ exports[`typegen generateRosLib works with zero topics or datatypes 1`] = `
...
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/__snapshots__/generateRosLib.test.ts.snap:336]:
```suggestion
     * Foxglove Studio session, so if a datatype changes, your User Script
     * may not compile on the newly formatted bag.
```

File: packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/generateRosLib.ts [hunk 6]
```diff
@@ -277,7 +277,7 @@ const generateRosLib = ({
      * Input<"/your_input_topic_2">'.
      *
      * These types are dynamically generated from the bag(s) currently in your
-     * Foxglove Studio session, so if a datatype changes, your Node Playground
+     * Foxglove Studio session, so if a datatype changes, your User Scripts
      * node may not compile on the newly formatted bag.
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/generateRosLib.ts:281]:
```suggestion
     * Foxglove Studio session, so if a datatype changes, your User Script
     * may not compile on the newly formatted bag.
```

File: packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/typescript/ros.ts [hunk 7]
```diff
@@ -36,7 +36,7 @@ export const ros_lib_dts = `
    * Input<"/your_input_topic_2">'.
    *
    * These types are dynamically generated from the bag(s) currently in your
-   * Foxglove Studio session, so if a datatype changes, your Node Playground
+   * Foxglove Studio session, so if a datatype changes, your User Scripts
    * node may not compile on the newly formatted bag.
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/typescript/ros.ts:40]:
```suggestion
   * Foxglove Studio session, so if a datatype changes, your User Script
   * may not compile on the newly formatted bag.
```

File: packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/typescript/templates/marker.ts.template [hunk 8]
```diff
@@ -1,6 +1,6 @@
-// This example shows how to publish a Marker message from a Node Playground node.
+// This example shows how to publish a Marker message from a User Script node.
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/nodeTransformerWorker/typescript/templates/marker.ts.template:1]:
```suggestion
// This example shows how to publish a Marker message from a User Script.
```

File: packages/studio-base/webpack.ts [hunk 9]
```diff
@@ -19,7 +19,7 @@ if (monacoPkg.version !== "0.30.1") {
   throw new Error(`
     It looks like you are trying to change the version of Monaco.
 
-    Please make a node playground node and confirm that loading a data source properly updates
+    Please make a user script node and confirm that loading a data source properly updates
```
Comment (amacneil) [inline packages/studio-base/webpack.ts:22]:
```suggestion
    Please make a User Script and confirm that loading a data source properly updates
```

File: packages/studio-base/src/players/UserNodePlayer/index.ts [hunk 10]
```diff
@@ -478,7 +478,7 @@ export default class UserNodePlayer implements Player {
 
         if (!result) {
           this._problemStore.set(problemKey, {
-            message: `Node playground node ${nodeId} timed out`,
+            message: `User script node ${nodeId} timed out`,
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/index.ts:481]:
```suggestion
            message: `User Script ${nodeId} timed out`,
```

File: packages/studio-base/src/players/UserNodePlayer/index.ts [hunk 11]
```diff
@@ -503,7 +503,7 @@ export default class UserNodePlayer implements Player {
         if (!result.message) {
           this._problemStore.set(problemKey, {
             severity: "warn",
-            message: `Node playground node ${nodeId} did not produce a message`,
+            message: `User script node ${nodeId} did not produce a message`,
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/index.ts:506]:
```suggestion
            message: `User Script ${nodeId} did not produce a message`,
```

File: packages/studio-base/src/players/UserNodePlayer/index.ts [hunk 12]
```diff
@@ -419,7 +419,7 @@ export default class UserNodePlayer implements Player {
 
               this._problemStore.set(problemKey, {
                 severity: "error",
-                message: `Node playground runtime error: ${String(event.data)}`,
+                message: `User scripts runtime error: ${String(event.data)}`,
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/index.ts:422]:
```suggestion
                message: `User script runtime error: ${String(event.data)}`,
```

File: packages/studio-base/src/players/UserNodePlayer/index.ts [hunk 13]
```diff
@@ -432,7 +432,7 @@ export default class UserNodePlayer implements Player {
 
               this._problemStore.set(problemKey, {
                 severity: "error",
-                message: `Node playground runtime error: ${msg}`,
+                message: `User scripts runtime error: ${msg}`,
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/index.ts:435]:
```suggestion
                message: `User script runtime error: ${msg}`,
```

File: packages/studio-base/src/players/UserNodePlayer/index.ts [hunk 14]
```diff
@@ -556,7 +556,7 @@ export default class UserNodePlayer implements Player {
 
         this._problemStore.set("worker-error", {
           severity: "error",
-          message: `Node playground error: ${event.message}`,
+          message: `User scripts error: ${event.message}`,
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/index.ts:559]:
```suggestion
          message: `User Script error: ${event.message}`,
```

File: packages/studio-base/src/players/UserNodePlayer/index.ts [hunk 15]
```diff
@@ -568,7 +568,7 @@ export default class UserNodePlayer implements Player {
 
         this._problemStore.set("worker-error", {
           severity: "error",
-          message: `Node playground error: ${String(event.data)}`,
+          message: `User scripts error: ${String(event.data)}`,
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/index.ts:571]:
```suggestion
          message: `User Script error: ${String(event.data)}`,
```

File: packages/studio-base/src/players/UserNodePlayer/index.ts [hunk 16]
```diff
@@ -581,7 +581,7 @@ export default class UserNodePlayer implements Player {
 
         this._problemStore.set("worker-error", {
           severity: "error",
-          message: `Node playground error: ${msg}`,
+          message: `User scripts error: ${msg}`,
```
Comment (amacneil) [inline packages/studio-base/src/players/UserNodePlayer/index.ts:584]:
```suggestion
          message: `User Script error: ${msg}`,
```

**Comment by defunctzombie** [comment]
image

A hover title still says "nodes" :D

**Comment by foxymiles** [comment]
> image
> 
> A hover title still says "nodes" :D

Good catch. Fixed.

**Review by defunctzombie — APPROVED** [review]

**Comment by foxymiles** [comment]
@esthersweon Can we coordinate on updating the docs for this name change?

**Comment by esthersweon** [comment]
@amacneil @foxymiles - website PR opened and ready to merge.

**Pull request merged**